src/classes/RD2_OpportunityEvaluationService_TEST.cls

```diff
@@ -238,17 +245,212 @@ private class RD2_OpportunityEvaluationService_TEST {
         System.assertEquals(false, oppById.get(oldOpps[1].Id).isClosed, 'Open Opp should stay open');
     }
 
+    /**
+    * @description When Installment Auto Creation Setting is set to Disable First Installment, 
+    * the installment should not be created on insert of RD
+    */
+    @IsTest
+    private static void shouldNotCreateOppOnRDInsertWhenDisableFirstInstallment() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Map<String, Object> rdSettingMap = new Map<String, Object> {
+            'InstallmentOppAutoCreateOption__c' => OPTION_NAME_DISABLE_FIRST
+        };
+
+        RD2_Settings_TEST.setUpConfiguration(rdSettingMap);
+        
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder(getContact()).build();
+        
+        Test.startTest();
+        insert rd;
+        Test.stopTest();
+
+        List<Opportunity> opps = oppGateway.getRecords(rd);
+
+        System.assertEquals(0, opps.size(),
+            'No installment opp should be created on insert of RD when the auto Creation setting is ' + OPTION_NAME_DISABLE_FIRST);
+    }
+
+    /**
+    * @description When Installment Auto Creation Setting is set to Disable First Installment, 
+    * the installment should be created on RD when the nightly batch job run
+    */
+    @IsTest
+    private static void shouldCreateOppDuringBatchRunWhenDisableFirstInstallment() {
```

Comment:
Let's move this test verifying the batch `RD2_OpportunityEvaluation_BATCH` into `RD2_OpportunityEvaluation_TEST`.